config/yaml-mp/src/main/java/io/helidon/config/yaml/mp/YamlConfigSourceProvider.java

```diff
@@ -21,30 +21,46 @@
 import java.util.Enumeration;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Objects;
 
 import io.helidon.config.ConfigException;
+import io.helidon.config.mp.spi.MpConfigSourceProvider;
 
 import org.eclipse.microprofile.config.spi.ConfigSource;
-import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
 
 /**
  * YAML config source provider for MicroProfile config that supports file {@code application.yaml}.
  * This class should not be used directly - it is loaded automatically by Java service loader.
  */
-public class YamlConfigSourceProvider implements ConfigSourceProvider {
+public class YamlConfigSourceProvider implements MpConfigSourceProvider {
+    private static final String RESOURCE_NAME = "application.yaml";
+
     @Override
     public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader) {
-        Enumeration<URL> resources;
-        try {
-            resources = classLoader.getResources("application.yaml");
-        } catch (IOException e) {
-            throw new ConfigException("Failed to read resources from classpath", e);
-        }
+        return configSources(classLoader, null);
+    }
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader, String profile) {
+        Objects.requireNonNull(profile, "Profile must not be null");
+        return configSources(classLoader, profile);
+    }
 
+    private Iterable<ConfigSource> configSources(ClassLoader classLoader, String profile) {
         List<ConfigSource> result = new LinkedList<>();
-        while (resources.hasMoreElements()) {
-            URL url = resources.nextElement();
-            result.add(YamlMpConfigSource.create(url));
+        if (profile == null) {
```

Comment:
Why not reuse YamlMpConfigSource.classPath(String resource) and write more concise?

maybe something like this? 

return Objects.isNull(profile)? YamlMpConfigSource.classPath(RESOURCE_NAME) : YamlMpConfigSource.classPath(RESOURCE_NAME, profile, classLoader)

Reply:
Because `YamlMpConfigSource.classPath(RESOURCE_NAME)` does not accept classLoader as an argument.

Reply:
yea, but why do you need to pass it? it already uses the same Thread.currentThread().getContextClassLoader() as classPathConfigSources(resource, profile, ...);

Reply:
Because getConfigSources() which was overriden from `io.helidon.config.mp.spi.MpConfigSourceProvider` and in turn inherited from `org.eclipse.microprofile.config.spi.ConfigSourceProvider` has classLoader in its API. It is possible that the implementation can be used in some other scenario where the classPath will not be `Thread.currentThread().getContextClassLoader()`.